### DISCUSSION TITLE
'Request' object with WebSockets

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [fastapi/fastapi#7054] [Questions] [answered]
It has caught my attention that the 'Request' object is not available for WebSockets routes. Am I doing something wrong, or is this expected behaviour?

I'm trying to inject a dependency that requires a 'Request' object to return the DB session.

**Comment by includeamin** [comment]
Hi,
What kind of data do you require from `Request`? header? token in the header?
``` python
@app.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket):
    print(websocket.headers)
    print(websocket.path_params)
    print(websocket.query_params)
    await websocket.accept()
    while True:
        data = await websocket.receive_text()
        await websocket.send_text(f"Message text was: {data}")
```
I think you can use `WebSocket` object for your use-case
[check this section](https://fastapi.tiangolo.com/advanced/websockets/#using-depends-and-others)

**Comment by WilliamDEdwards** [comment]
I'd like to use the Request's state to get my DB session. But I now see middleware only supports HTTP, and that's where I do `request.state.db = SessionLocal()`...

**Comment by includeamin** [comment, marked as answer]
``` python
websocket.state
```

**Comment by Mause** [comment]
I would recommend that you consider how long your websocket will be open for as well, as your database connection will be open for the same length of time if you use that technique

**Comment by agordhandas** [comment]
When I needed to access the db connection in websocket operations, I ended up opening and closing the connection as needed:

```
db  = SessionLocal()
…
…
db.close()
```

**Comment by WilliamDEdwards** [comment]
Closing. Using the `WebSocket` object now.

**Comment by Karoid** [comment]
@WilliamDEdwards Can you tell me the solution with more detail? 
I am having same trouble with using request.status.db inside websocket.

**Comment by WilliamDEdwards** [comment]
> @WilliamDEdwards Can you tell me the solution with more detail? I am having same trouble with using request.status.db inside websocket.

Your database session probably won't be valid for as long as your WebSocket is open.
